### PR TITLE
CSP - Added support of get and set last mirror run

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://github.com/demisto/etc/issues/44845)

## Description
Added the 2 new functions:
1. `getLastMirrorRun`
2. `setLastMirrorRun`

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -8273,6 +8296,34 @@ def set_feed_last_run(last_run_indicators):
...
+    Before XSOAR version 6.6.0: we don't set the given data`.
+    :type last_mirror_run: ``dict``
+    :param last_mirror_run: Data to save in "LastMirrorRun" object.
+    :rtype: ``None``
+    :return: None
+    """
+    if is_demisto_version_ge('6.6.0'):
+        demisto.setLastMirrorRun(last_mirror_run)
+    else:
+        return "You cannot use setLastMirrorRun as your version is below 6.6.0"
```
Comment (abaumgarten) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8311]: I am not sure about the else. @anara123 what do you say?

Comment (anara123) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8307]: Keep it. This is ok this way

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -4260,18 +4263,18 @@ def is_valid_type(_type):
...
-            p=None,  # type: str
-            q=None,  # type: str
-            g=None,  # type: str
-            modulus=None,  # type: str
-            exponent=None,  # type: int
-            x=None,  # type: str
-            y=None,  # type: str
-            curve=None  # type: str
+                self,
+                algorithm,  # type: str
```
Comment (anara123) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:4267]: Revert all these indentations. I don't think it was intended

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 1 above)
Comment (anara123) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8307]: raise DemistoException instead, don't return string.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -8273,6 +8292,34 @@ def set_feed_last_run(last_run_indicators):
         demisto.setIntegrationContext(last_run_indicators)
 
 
+def set_last_mirror_run(last_mirror_run):
```
Comment (anara123) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8295]: Add typing.
like 
`# type: (Any, Optional[str], bool) -> Optional[int]`
it is supported in python2

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 2 above)
Comment (abaumgarten) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py]: Done

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 3 above)
Comment (abaumgarten) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py]: Done

**Review by guykeller — APPROVED** [review]
Looks good IMO, great job!

**Review by anara123 — APPROVED** [review]

**Pull request merged**